app/javascript/components/utils/NavBar.jsx

```diff
@@ -1,41 +1,166 @@
+    this.isOrganization = this.props.organization ? 1 : 0;
+    this.isVolunteer = this.props.user ? 1 : 0; 
+    this.state = {
+      listOpen: false
+    }
+    axios.defaults.headers.common = {
+      "X-Requested-With": "XMLHttpRequest",
+      "X-CSRF-TOKEN": document
+        .querySelector('meta[name="csrf-token"]')
+        .getAttribute("content")
+    };
   }
 
   goToDashboard = () => {
     window.location.href = `/`;
   };
 
   goToProfile = () => {
-    window.location.href = `/users/${this.props.user.id}`;
-  };
+    if (this.isOrganization) {
+      window.location.href = `/organizations/${this.props.organization.id}`;
+    } else if (this.isVolunteer) {
+      window.location.href = `/users/${this.props.user.id}`;
+    } else {
+      window.location.href = `/`;
+    }
+  }
 
   goToBrowse = () => {
     window.location.href = `/projects`;
   };
 
+  goToMyProjects = e => {
+    e.preventDefault();
+    window.location.href = '/my-projects';
+  };
+
+  goToApplications = (e) => {
+    e.preventDefault();
+    window.location = "/applications";
+  };
+
+  handleLogout = e => {
+    e.preventDefault();
+    if (this.isOrganization) {
+      axios
+      .delete("/organizations/sign_out", {})
+      .then(function(response) {
+        window.location.href = "/";
+      })
+      .catch(function(error) {
+        console.log(error);
+      });
+    } else if (this.isVolunteer) {
+      axios
+      .delete("/users/sign_out", {})
+      .then(function(response) {
+        window.location.href = "/";
+      })
+      .catch(function(error) {
+        console.log(error);
+      });
+    } else {
+      console.log("not signed in");
+    }
+  };
+
+  handleClickOutside(e) {
+    this.setState({
+        listOpen: false
+    });
+  }
+
+  toggleList() {
+    this.setState(prevState => ({
+        listOpen: !prevState.listOpen
+    }));
+  }
+
+  renderProfile() {
+    let profile = null;
+    if (this.isOrganization) {
+    profile = <li className="fr f4 tl dropdown"> 
+                <a className="f4 black" onClick={() => this.toggleList()}>
+                  {this.props.organization.name} <i className="fa fa-caret-down ml1"></i>
+                </a>
+                {this.state.listOpen ?
+                  <div className="ba dropdown-content h-auto">
+                    <a onClick={this.goToProfile}>Profile</a>
+                    <a className="bt pt2" onClick={this.goToMyProjects}>My Projects</a>
+                    <a className="bt pt2" onClick={this.goToApplications}>Applications</a>
+                    <a className="bt pt2 pb2" onClick={this.handleLogout}>Logout</a>
+                  </div>:null
+                }
+              </li>
+    } else if (this.isVolunteer) {
+      profile = <li className="fr f4 tl dropdown"> 
+                  <a className="f4 black" onClick={() => this.toggleList()}>
+                    {this.props.user.first_name} <i className="fa fa-caret-down ml1"></i>
```

Comment:
It will be cleaner if you put a conditional on the name portion only, instead of altering the entire profile block since much of it will be the same. You can have something inline like: `{this.props.user ? this.props.user.first_name + " " + this.props.user.last_name : this.props.organization.name}`.

I see that in the case that neither of them are logged in, then it does something else. But in that case, we might will prob not need the navbar since they will just be redirected to our login/signup page.